libcloud/compute/drivers/libvirt_driver.py

```diff
@@ -257,13 +257,27 @@ def _get_ip_addresses_for_domain(self, domain):
             # Only Linux is supported atm
             return result
 
+        if '///' not in self._uri:
+            # Only local libvirtd is supported atm
+            return result
+
         mac_addresses = self._get_mac_addresses_for_domain(domain=domain)
 
-        cmd = ['arp', '-an']
-        child = subprocess.Popen(cmd, stdout=subprocess.PIPE,
-                                 stderr=subprocess.PIPE)
-        stdout, _ = child.communicate()
-        arp_table = self._parse_arp_table(arp_output=stdout)
+        arp_table = {}
+        try:
+            cmd = ['arp', '-an']
+            child = subprocess.Popen(cmd, stdout=subprocess.PIPE,
```

Comment:
Minor improvement which would also make it easier to test separately would be to refactor this into two methods (e.g. `_get_arp_table_using_arp_cmd` and `_get_arp_table_using_ip_cmd` or something along those lines).

Anyways, not a blocker.